demisto_sdk/commands/generate_unit_tests/generate_unit_tests.py

```diff
@@ -192,52 +195,53 @@ def solve(self, generator: UnitTestsGenerator) -> TestModule:
 def run_generate_unit_tests(**kwargs):
     global logger
     input_path = kwargs.get('input_path', '')
-    test_data_path = kwargs.get('test_data_path', '')
     output_dir = kwargs.get('output_dir', '')
     commands = arg_to_list(kwargs.get('commands', ''))
     commands_examples_path = kwargs.get('examples', '')
     insecure = kwargs.get('insecure', False)
     use_demisto = kwargs.get('use_demisto', False)
+    append = kwargs.get('append', False)
 
     click.echo("================= Running Unit Testing Generator ===================")
     # validate inputs
+    input_path_obj = Path(input_path)
     if not input_path:
         print_error(
             'To use the generate_unit_tests version of this command please include an `input` argument')
         return 1
 
-    if input_path and not os.path.isfile(input_path):
+    if input_path and not input_path_obj.is_file():
         print_error(F'Input file {input_path} was not found.')
         return 1
 
     if not input_path.lower().endswith('.py'):
         print_error(F'Input {input_path} is not a valid python file.')
         return 1
 
-    module_name = os.path.basename(input_path)
+    module_name = input_path_obj.name
 
-    if not test_data_path:
-        dirname = os.path.dirname(input_path)
-        test_data_path = os.path.join(f'{dirname}/', "test_data")
-        if not os.path.isdir(test_data_path):
-            print_error(
-                'There is no test_data folder in the working directory, please insert test data directory path.')
-            return 1
+    dirname = input_path_obj.parent
+    test_data_path = Path(dirname, 'test_data')
+    if not test_data_path.is_dir():
+        print_error(
+            'There is no test_data folder in the working directory, please insert test data directory path.')
+        return 1
 
     if not output_dir:
-        output_dir = os.path.dirname(input_path)
+        output_dir = Path(input_path).parent
 
     if not commands_examples_path:
-        commands_examples_path = os.path.join(os.path.dirname(input_path), 'command_examples')
+        commands_examples_path = Path(input_path_obj.parent, 'command_examples')
 
     # Check the directory exists and if not, try to create it
-    if not os.path.exists(output_dir):
+    output_dir_path_obj = Path(output_dir)
+    if not output_dir_path_obj.exists():
```

Comment:
why not `if not output_dir.exists()`?
btw, `Path` objects have `.mkdir`, better use it over mixing Pathlib & OS

Reply:
output_dir may be string at that point

Reply:
I'd suggest converting it to `Path` as soon as it's set